File: frontend/src/types/dictionary.ts
export type DictionaryItem = {
  id: number;
  code: string;
  name: string;
};

export type GradeItem = {
  id: number;
  grade_index: number;
  grade_code?: string | null;
  grade_name: string;
  edu_stage?: string | null;
};

export type CognitiveLevelItem = {
  id: number;
  code: string;
  name: string;
  level_order: number;
};

export type CompetencyItem = {
  id: number;
  code: string;
  name: string;
  display_order: number;
};

File: backend/app/schemas/dictionary.py
from pydantic import BaseModel, ConfigDict


class DictionaryItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    code: str
    name: str


class GradeItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    grade_index: int
    grade_code: str | None = None
    grade_name: str
    edu_stage: str | None = None


class KnowledgeTypeItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    source_type_code: str
    source_type_name: str


class CognitiveLevelItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    code: str
    name: str
    level_order: int


class CompetencyItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    code: str
    name: str
    display_order: int
